crates/core/tedge_agent/src/main.rs

```diff
@@ -20,17 +20,33 @@ pub struct AgentOpt {
     /// If on also reports DEBUG and TRACE
     #[structopt(long)]
     pub debug: bool,
+
+    /// Start the agent with clean session off, subscribe to the topics, so that no messages are lost
+    #[structopt(short, long)]
+    pub init: bool,
+
+    /// Start the agent with clean session on, drop the previous session and subscriptions
+    ///
+    /// WARNING: All pending messages will be lost.
+    #[structopt(short, long)]
+    pub clear: bool,
 }
 
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error> {
-    let agent = AgentOpt::from_args();
-    tedge_utils::logging::initialise_tracing_subscriber(agent.debug);
+    let agent_opt = AgentOpt::from_args();
     let tedge_config_location = tedge_config::TEdgeConfigLocation::from_default_system_location();
     let mut agent = agent::SmAgent::try_new(
         "tedge_agent",
         SmAgentConfig::try_new(tedge_config_location)?,
     )?;
-    agent.start().await?;
+    tedge_utils::logging::initialise_tracing_subscriber(agent_opt.debug);
```

Comment:
I would do this at the very beginning as done before.

Reply:
fixed